client/marshaller/kryo-marshaller/src/main/java/org/infinispan/marshaller/kryo/KryoTranscoder.java

```diff
@@ -0,0 +1,55 @@
+package org.infinispan.marshaller.kryo;
+
+import net.spy.memcached.CachedData;
+import net.spy.memcached.transcoders.Transcoder;
+
+/**
+ * @author Ryan Emerson
+ * @since 9.0
+ */
+public class KryoTranscoder implements Transcoder<Object> {
+
+   private KryoMarshaller marshaller;
+
+   public KryoTranscoder(KryoMarshaller marshaller) {
+      this.marshaller = marshaller;
+   }
+
+   public KryoMarshaller getMarshaller() {
+      return marshaller;
+   }
+
+   public void setMarshaller(KryoMarshaller marshaller) {
+      this.marshaller = marshaller;
+   }
+
+   @Override
+   public boolean asyncDecode(CachedData d) {
+      return false;
+   }
+
+   @Override
+   public CachedData encode(Object o) {
+      try {
+         byte[] bytes = marshaller.objectToByteBuffer(o);
+         return new CachedData(0, bytes, bytes.length);
+      } catch (Throwable t) {
+         throw new RuntimeException(t);
```

Comment:
The exceptions should be transformed into more meaningful ones, at least `CacheException`, and they should be fired through `Log` object, see examples in code.